extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -255,7 +258,7 @@ class QueueProcessor extends EventEmitter {
             });
             this._consumer.on('metrics', data => {
                 // i.e. data = { my-site: { ops: 1, bytes: 124 } }
-                this._mProducer.publishMetrics(data,
+                this._mProducer.publishMetrics(data[this.site],
```

Comment:
Did you change the API of `publishMetrics`? It seems to me you changed the structure of metrics passed by accessing `[this.site]` key, and I would think that each specialized processor would push metrics the same way but will only provide metrics for the site they're responsible for as a consequence.

Reply:
The constructor change should be reflected in `ReplicateObject` and site is passed through the `getStateVars`